feat(core): enhance profiling with documentation URLs

SkyZeroZx opened the pull request:
Enhances the Chrome DevTools performance profiling integration by adding links to relevant Angular documentation for lifecycle hooks and profiler events.

Fixes #63959

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/contributing-docs/commit-message-guidelines.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.dev application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
 Currently, no link to documentation is displayed.

## What is the new behavior?

Validation is added to use performance API instead of console.timestap (extended), since it does not support adding custom data such as links according to the documentation and some tests I performed.

See https://developer.chrome.com/docs/devtools/performance/extension#key_benefits

<img width="690" height="710" alt="image" src="https://github.com/user-attachments/assets/d7d05d16-49aa-4bf2-9fc6-f7cbec8450b6" />


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

JeanMeche commented on packages/core/src/render3/debug/chrome_dev_tools_performance.ts:
```diff
@@ -46,13 +46,103 @@ declare global {
+    ngafterviewinit: `${DOC_URL_LIFECYCLE_HOOKS}#ngafterviewinit`,
+    ngafterviewchecked: `${DOC_URL_LIFECYCLE_HOOKS}#ngafterviewchecked`,
+    ngondestroy: `${DOC_URL_LIFECYCLE_HOOKS}#ngondestroy`,
+  };
```
we can extract that const outside of the function

SkyZeroZx replied:
Updated

dgp1130 commented on packages/core/src/render3/debug/chrome_dev_tools_performance.ts:
```diff
@@ -46,13 +46,103 @@ declare global {
+const TRACK_NAME = '\u{1F170}\uFE0F Angular';
+
+function timestampMark(markName: string): void {
+  if (hasPerformanceMeasure) {
```
@pkozlowski-opensource, I assume it's ok to switch to `performance.mark` but I'm not very familiar with the nuances vs `console.timeStamp` and figured we should check with you.

SkyZeroZx commented on packages/core/src/render3/debug/chrome_dev_tools_performance.ts:
```diff
@@ -32,19 +32,27 @@ type DevToolsColor =
+const TRACK_NAME = '\u{1F170}\uFE0F Angular';
+
+const isPreRelease =
+  VERSION.full.includes('-next') || VERSION.full.includes('-rc') || VERSION.full === '0.0.0';
```
Done

SkyZeroZx committed:
feat(core): enhance profiling with documentation URLs

Enhances the Chrome DevTools performance profiling integration by adding links to relevant Angular documentation for lifecycle hooks and profiler events.

Fixes #63959

dgp1130 reviewed (approved):
LGTM, as long as @pkozlowski-opensource doesn't have any concerns with switching to `performance.mark`, we should be good to go.

Thanks for putting this together!
